### PR TITLE
Comment PR with test results

### PR DESCRIPTION
**Pull request by bayandin**
## Describe your changes

This PR adds posting a comment with test results. Each workflow run updates the comment with new results.
The layout and the information that we post can be changed for our need, right now it contains failed tests and test which changes status after rerun (i.e. flaky tests)

Here's an example of a comment https://github.com/neondatabase/neon/pull/3907#issuecomment-1489361853

## Issue ticket number and link
N/A

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

### CODE REVIEW

File: scripts/pr-comment-test-report.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+//         fetch,
+//         reportUrlDebug: "...",
+//         reportUrlRelease: "...",
+//       })
+
+module.exports = async ({ github, context, fetch, reportUrlDebug, reportUrlRelease }) => {
+    // Marker to find the comment in the subsequent runs
+    const startMarker = `<!--AUTOMATIC COMMENT START #${context.payload.number}-->`
+    // GitHub bot id taken from (https://api.github.com/users/github-actions[bot])
+    const githubActionsBotId = 41898282
```
Comment (vadim2404) [inline scripts/pr-comment-test-report.js:19]: should it be hidden under github vars?

File: scripts/pr-comment-test-report.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        owner: context.repo.owner,
+        repo: context.repo.repo,
+    }
+
+    for (const reportUrl of [reportUrlDebug, reportUrlRelease]) {
+        if (!reportUrl) {
+            continue
+        }
+
+        const buildType = (reportUrl === reportUrlDebug) ? "Debug" : "Release"
```
Comment (vadim2404) [inline scripts/pr-comment-test-report.js:36]: I'd prefer more structural way

```js
const reportUrls = [{
  type: "debug",
  url: "<<debug_url>>",
}, {
  type: "release",
  url: "<<release_url>>",
}]
```

**Review by SomeoneToIgnore — COMMENTED** [review]
The JS part is looks very magical, I wonder, if any public GH Actions exist to do similar things? 

Otherwise, LGTM and works, wow.
I would need to add a [compilation timings report](https://github.com/neondatabase/neon/blob/bf46237fc22800625cf86578ca9027bdfa047d19/.github/workflows/neon_extra_builds.yml#L138) as a comment too, thank you for the idea.

File: .github/actions/allure-report/action.yml [hunk 3]
```diff
@@ -15,8 +15,16 @@ outputs:
...
     value: ${{ steps.generate-report.outputs.report-url }}
+  report-json-url:
+    description: 'Allure report JSON URL'
+    value: ${{ steps.generate-report.outputs.report-json-url }}
 
 runs:
+  env:
+    BUILD_TYPE: ${{ inputs.build_type }}
+    BUCKET: neon-github-public-dev
+    TEST_OUTPUT: /tmp/test_output
```
Comment (bayandin) [inline .github/actions/allure-report/action.yml:26]: Weird, it looks like GitHub doesn't like such syntax:
```
Error: /nvme/actions-runner/_work/neon/neon/./.github/actions/allure-report/action.yml (Line: 29, Col: 3): Unexpected value 'steps'
Error: System.ArgumentNullException: Value cannot be null. (Parameter 'You are using a composite action but there are no steps provided in /nvme/actions-runner/_work/neon/neon/./.github/actions/allure-report/action.yml.')
   at GitHub.Runner.Worker.ActionManifestManager.ConvertRuns(IExecutionContext executionContext, TemplateContext templateContext, TemplateToken inputsToken, String fileRelativePath, MappingToken outputs)
   at GitHub.Runner.Worker.ActionManifestManager.Load(IExecutionContext executionContext, String manifestFile)
```

Found similar issues in https://github.com/orgs/community/discussions/51280

**Review by vadim2404 — APPROVED** [review]

**Review by SomeoneToIgnore — APPROVED** [review]

**Pull request merged**